src/components/primitive/avatar/avatar.component.tsx

```diff
@@ -0,0 +1,55 @@
+import styled, { withTheme } from 'styled-components';
+import { AvatarComponentProps, AvatarSize, Variant } from './avatar.component.props';
+import { Small, Medium, Large } from './avatar.component.styles';
+import { IStyledTheme } from '../../../themes/styled-components';
+
+export const Avatar: React.FunctionComponent<AvatarComponentProps> = withTheme(
+  (props: AvatarComponentProps & IStyledTheme) => {
+    const {
+      variant,
+      name,
+      src,
+      theme: { images },
+      ...rest
+    } = props;
+
+    /**
+     * Fallback image.
+     */
+    const url = name ? images[name] : src;
+
+    /**
+     * @param variant - variant passed by the user.
```

Comment:
Code comments for function params generally do not include who passed them. It must be about what those params are, and what the function is expecting.

Reply:
Also, you need not write comments for non exported functions unless they are too complex to understand. Whereas it's good to have proper comments for all the exposed ones.